module-services/service-cellular/ServiceCellular.cpp

```diff
@@ -1075,7 +1081,20 @@ auto ServiceCellular::receiveSMS(std::string messageNumber) -> bool
                 if (messageParsed) {
                     messageParsed = false;
 
-                    const auto decodedMessage = UCS2(messageRawBody).toUTF8();
+                    UTF8 decodedMessage;
+
+                    const std::string decodedStr  = utils::hexToBytes<std::string>(messageRawBody);
+                    const auto mmsNotificationOpt = pdu::parse(decodedStr);
+                    if (mmsNotificationOpt) {
+                        std::string number = numberFromAddress(mmsNotificationOpt->fromAddress);
+                        // NOTE: number may be empty
+                        decodedMessage = UTF8("[MMS]");
```

Comment:
wondering if MMS in other languages has different spelling - if so, we'd need to put it through our translation system

Reply:
Yes, it seems there is. I wanted to avoid localization but since we have to do this anyway we could as well think about more elaborate message, e.g.: `MMS not supported`.

Reply:
However AFAIS in all languages that we support it's called MMS. Just like it's SMS in all supported languages. So I think it can stay as it is. I'm not an expert though.

Reply:
fine ;)